src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/NVidiaEmbeddingClient.java

```diff
@@ -102,8 +107,17 @@ public Uni<List<float[]>> vectorize(
         .onItem()
         .transform(
             resp -> {
+              validateEmbeddingResponse(resp);
               Arrays.sort(resp.data(), (a, b) -> a.index() - b.index());
               return Arrays.stream(resp.data()).map(data -> data.embedding()).toList();
             });
   }
+
+  private void validateEmbeddingResponse(EmbeddingResponse response) {
+    // if there is no data return
+    if (response.data() == null || response.data().length == 0) {
+      throw EMBEDDING_PROVIDER_INVALID_RESPONSE.toApiException(
+          "No data return from the embedding provider");
+    }
+  }
```

Comment:
Only add deserialized Json validation for Nvdia. Will create another PR to validate all other providers' deserialized response

Reply:
Not sure if this even needed, since we have checked for required number of embedding response in DataVectorizer. May be to check for null and return empty list is good enough here.